src/libs/WorkflowUtils.ts

```diff
@@ -0,0 +1,162 @@
+            break;
+        }
+        currentApproverEmails.add(nextEmail);
+
+        // If there is a forwardsTo, set the next approver to the forwardsTo
+        nextEmail = employees[nextEmail].forwardsTo;
+    }
+
+    return approvers;
+}
+
+type ConvertPolicyEmployeesToApprovalWorkflowsParams = {
+    /**
+     * List of employees in the policy
+     */
+    employees: PolicyEmployeeList;
+
+    /**
+     * Personal details of the employees
+     */
+    personalDetails: PersonalDetailsList;
+
+    /**
+     * Email of the default approver for the policy
+     */
+    defaultApprover: string;
+};
+
+/** Convert a list of policy employees to a list of approval workflows */
+function convertPolicyEmployeesToApprovalWorkflows({employees, defaultApprover, personalDetails}: ConvertPolicyEmployeesToApprovalWorkflowsParams): ApprovalWorkflow[] {
+    const approvalWorkflows: Record<string, ApprovalWorkflow> = {};
+
+    // Keep track of how many times each approver is used to detect approvers in multiple workflows
+    const approverCountsByEmail: Record<string, number> = {};
+    const personalDetailsByEmail = lodashMapKeys(personalDetails, (value, key) => value?.login ?? key);
+
+    // Add each employee to the appropriate workflow
+    Object.values(employees).forEach((employee) => {
+        const {email, submitsTo} = employee;
+        if (!email || !submitsTo) {
+            return;
+        }
+
+        const member: Member = {email, avatar: personalDetailsByEmail[email]?.avatar, displayName: personalDetailsByEmail[email]?.displayName ?? email};
+        if (!approvalWorkflows[submitsTo]) {
+            const approvers = getApprovalWorkflowApprovers({employees, firstEmail: submitsTo, personalDetailsByEmail});
+            approvers.forEach((approver) => (approverCountsByEmail[approver.email] = (approverCountsByEmail[approver.email] ?? 0) + 1));
+
+            approvalWorkflows[submitsTo] = {
+                members: [],
+                approvers,
+                isDefault: defaultApprover === submitsTo,
+                isBeingEdited: false,
+            };
+        }
+        approvalWorkflows[submitsTo].members.push(member);
+    });
+
+    // Sort the workflows by the first approver's name (default workflow has priority)
+    const sortedApprovalWorkflows = Object.values(approvalWorkflows).sort((a, b) => {
+        if (a.isDefault) {
+            return -1;
+        }
+
+        return (a.approvers[0]?.displayName ?? '-1').localeCompare(b.approvers[0]?.displayName ?? '-1');
+    });
+
+    // Add a flag to each approver to indicate if they are in multiple workflows
+    return sortedApprovalWorkflows.map((workflow) => ({
+        ...workflow,
+        approvers: workflow.approvers.map((approver) => ({
+            ...approver,
+            isInMultipleWorkflows: approverCountsByEmail[approver.email] > 1,
+        })),
+    }));
+}
+
+type ConvertApprovalWorkflowToPolicyEmployeesParams = {
+    approvalWorkflow: ApprovalWorkflow;
```

Comment:
NAB: It would be nice to see the `/**/` comments on these too, like you did above.